saleor/graphql/warehouse/filters.py

```diff
@@ -0,0 +1,40 @@
+import django_filters
+
+from ...warehouse.models import Warehouse
+from ..core.types import FilterInputObjectType
+from ..utils import filter_by_query_param
+
+
+def prefech_qs_for_filter(qs):
+    return qs.select_related("address")
+
+
+def filter_search(qs, _, value):
+    search_fields = [
+        "name",
+        "company_name",
+        "email",
+        "address__street_address_1",
+        "address__street_address_2",
+        "address__city",
+        "address__postal_code",
+        "address__phone",
+    ]
+
+    if value:
+        qs = prefech_qs_for_filter(qs)
+        qs = filter_by_query_param(qs, value, search_fields)
+    return qs
+
+
+class WarehouseFilter(django_filters.FilterSet):
+    search = django_filters.CharFilter(method=filter_search)
+
+    class Meta:
+        model = Warehouse
+        fields = []
```

Comment:
`fields = ['search']` or AFAIK you could remove this line

Reply:
So for a model, we must define either `fields` or `exclude`. But I'm not sure it's a good idea to set it to `search` as this is not one of model field. @maarcingebala  what do you think?

Reply:
If there are no other filters than `search` I think this should be empty. If I'm correct it's mandatory to have either `fields` or `exclude`.